packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,384 @@
+        require(_dsi.inputSet(), "DeploySuperchain: Input not set");
+        address guardian = _dsi.guardian();
+        bool paused = _dsi.paused();
+
+        ProxyAdmin superchainProxyAdmin = _dso.superchainProxyAdmin();
+        SuperchainConfig superchainConfigImpl = _dso.superchainConfigImpl();
+        assertValidContractAddress(address(superchainProxyAdmin));
+        assertValidContractAddress(address(superchainConfigImpl));
+
+        vm.startBroadcast(msg.sender);
+        SuperchainConfig superchainConfigProxy = SuperchainConfig(address(new Proxy(address(superchainProxyAdmin))));
+        superchainProxyAdmin.upgradeAndCall(
+            payable(address(superchainConfigProxy)),
+            address(superchainConfigImpl),
+            abi.encodeCall(SuperchainConfig.initialize, (guardian, paused))
+        );
+        vm.stopBroadcast();
+
+        vm.label(address(superchainConfigProxy), "SuperchainConfigProxy");
+        _dso.set(_dso.superchainConfigProxy.selector, address(superchainConfigProxy));
+    }
+
+    function deployAndInitializeProtocolVersions(DeploySuperchainInput _dsi, DeploySuperchainOutput _dso) public {
+        require(_dsi.inputSet(), "DeploySuperchain: Input not set");
+
+        address protocolVersionsOwner = _dsi.protocolVersionsOwner();
+        ProtocolVersion requiredProtocolVersion = _dsi.requiredProtocolVersion();
+        ProtocolVersion recommendedProtocolVersion = _dsi.recommendedProtocolVersion();
+
+        ProxyAdmin superchainProxyAdmin = _dso.superchainProxyAdmin();
+        ProtocolVersions protocolVersionsImpl = _dso.protocolVersionsImpl();
+        assertValidContractAddress(address(superchainProxyAdmin));
+        assertValidContractAddress(address(protocolVersionsImpl));
+
+        vm.startBroadcast(msg.sender);
+        ProtocolVersions protocolVersionsProxy = ProtocolVersions(address(new Proxy(address(superchainProxyAdmin))));
+        superchainProxyAdmin.upgradeAndCall(
+            payable(address(protocolVersionsProxy)),
+            address(protocolVersionsImpl),
+            abi.encodeCall(
+                ProtocolVersions.initialize,
+                (protocolVersionsOwner, requiredProtocolVersion, recommendedProtocolVersion)
+            )
+        );
+        vm.stopBroadcast();
+
+        vm.label(address(protocolVersionsProxy), "ProtocolVersionsProxy");
+        _dso.set(_dso.protocolVersionsProxy.selector, address(protocolVersionsProxy));
+    }
+
+    function transferProxyAdminOwnership(DeploySuperchainInput _dsi, DeploySuperchainOutput _dso) public {
+        require(_dsi.inputSet(), "DeploySuperchain: Input not set");
+        address proxyAdminOwner = _dsi.proxyAdminOwner();
+
+        ProxyAdmin superchainProxyAdmin = _dso.superchainProxyAdmin();
+        assertValidContractAddress(address(superchainProxyAdmin));
+
+        vm.broadcast(msg.sender);
+        superchainProxyAdmin.transferOwnership(proxyAdminOwner);
+    }
+
+    // -------- Utilities --------
```

Comment:
This would be great to move into a `library` later, for reuse between scripts.

Reply:
Agreed, I am going to do that in a follow up PR once I have a better idea of what utils we want to reuse